src/components/common/CodeReviewTools.js

```diff
@@ -0,0 +1,401 @@
+            justify="flex-end"
+          >
+            {showEditButton ? (
+              <Grid item>
+                <Tooltip title={t('Edit')}>
+                  <FormControl>{editButtonControl}</FormControl>
+                </Tooltip>
+              </Grid>
+            ) : null}
+            {showVisibilityButton ? (
+              <Grid item>
+                <Tooltip
+                  title={
+                    allVisibleState
+                      ? t('Hide All Comments')
+                      : t('Show All Comments')
+                  }
+                >
+                  <FormControl>{hideAllCommentsToggleControl}</FormControl>
+                </Tooltip>
+              </Grid>
+            ) : null}
+          </Grid>
+        </Grid>
+        <CommitInfoDialog />
+      </>
+    );
+  };
+
+  render() {
+    const { topBarVisible } = this.props;
+    return topBarVisible ? this.renderTopBar() : null;
+  }
+}
+
+const mapStateToProps = (
+  { appInstance, appInstanceResources, users, layout, context },
+  ownProps,
+) => {
+  const { t } = ownProps;
+  const { lang } = context;
+  const codeSamples = appInstanceResources.content.filter(
+    (r) => r.type === CODE,
+  );
+  const instructorContributor = {
+    label: t('Instructor'),
+    value: DEFAULT_CODE_ID,
+  };
+  // get unique id of users that contributed code
+  const uniqueCodeContributors = [...new Set(codeSamples.map((r) => r.user))];
+  const codeContributors = uniqueCodeContributors.map((contributorName) => ({
+    label:
+      users.content.find((u) => u.id === contributorName)?.name || DEFAULT_USER,
+    value: contributorName,
+  }));
+  // build an array of: [{code, codeId, commitMessage, userName}]
+  const codeVersions = codeContributors.map((u) =>
+    codeSamples
+      .filter((r) => r.user === u.value)
+      .map((c) => {
+        const { commitMessage } = c.data;
+        const msg = `${commitMessage.slice(
+          0,
+          DEFAULT_TRUNCATION_COMMIT_MESSAGE_LENGTH,
+        )}${
+          commitMessage.length > DEFAULT_TRUNCATION_COMMIT_MESSAGE_LENGTH
+            ? '...'
+            : ''
+        }`;
+        const { createdAt = PLACEHOLDER_DATE } = c;
+        const date = formatDistance(Date.parse(createdAt), new Date(), {
+          addSuffix: true, // adds "ago" at the end
+          locale: locales[lang],
+        });
+        return {
+          label: `${msg} - ${date}`,
+          value: {
+            code: c.data.code,
+            codeId: c._id,
+          },
+        };
+      }),
+  );
+  const instructorCode = {
+    label: t('Default Version'),
+    value: {
+      code: appInstance.content.settings.code,
+      codeId: DEFAULT_CODE_ID,
+    },
+  };
+
+  return {
+    codeEditorSettings: layout.codeEditorSettings,
+    topBarVisible: appInstance.content.settings.topBarVisible,
+    codeContributors: [instructorContributor, ...codeContributors],
+    codeVersions: [[instructorCode], ...codeVersions],
```

Comment:
Why is `instructorCode` in an array?

Reply:
Code versions is a list of lists, first degree indexes contributors and the second degree is the individual versions of a single contributor. To respect this convention for the instructor i had to put it into an array even if there is by default only 1 version for instructor.

Reply:
Interesting. Maybe we can actually allow instructors to have many versions in the future. But good point!

Reply:
Yes that would be really interesting, i think that is why i tried to keep this format.